.github/workflows/ci-gh.yml

```diff
@@ -0,0 +1,57 @@
+name: CI (GPU)
+
+concurrency:
+  group: ci-gpu-on-${{ github.event_name }}-from-${{ github.ref_name }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches-ignore:
+      - gh-pages  # deployment target branch (this workflow should not exist on that branch anyway)
+  pull_request:
+    branches-ignore:
+      - gh-pages  # deployment target branch (this workflow should not exist on that branch anyway)
+
+jobs:
+  build:
+    permissions: read-all
+    runs-on: ubuntu-latest
+    container:
+      options: -u root
+      image: rapidsai/devcontainers:23.06-cpp-cuda11.8-mambaforge-ubuntu22.04
+      env:
+        PYTHONDONTWRITEBYTECODE: 1
+        SCCACHE_REGION: us-east-2
+        SCCACHE_BUCKET: rapids-sccache-devs
+        SCCACHE_S3_KEY_PREFIX: legate-cunumeric-dev
+        VAULT_HOST: https://vault.ops.k8s.rapids.ai
+        GH_TOKEN: "${{ secrets.PERSONAL_ACCESS_TOKEN }}"
+        GITHUB_TOKEN: "${{ secrets.PERSONAL_ACCESS_TOKEN }}"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        
+      - name: Build
+        shell: bash -eo pipefail {0}
+        run: |
+          apt update;
+          DEBIAN_FRONTEND=noninteractive \
+          apt install -y --no-install-recommends \
+              numactl rlwrap;
+          mv scripts/home/coder/.gitconfig /home/coder/.gitconfig;
```

Comment:
Should we add the scripts to this repository or keep it in a separate repository? Also, we probably don't want to use the scripts as is since they check out a version of legate core with `clone-legate-core`. Is this something that you plan to change as WIP on this PR?

Reply:
Yes I want to replace all uses of GitHub CLI with GH actions.

Reply:
> hould we add the scripts to this repository or keep it in a separate repository?

I don't mind keeping it in a separate repository. Anybody working on this in the future will need to be able to access this other repository.

Reply:
Probably, as we adjust the scripts to the usage in this repo, we could keep them here.

Reply:
Scripts into respective repository.